metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/ExecutionAggregator.java

```diff
@@ -1,56 +1,108 @@
 
     /**
-     * Returns a list of executions of type T where the execution metric is not null.
+     * Returns the property name of the metric in the Execution class to validate.
+     * This is used to determine if a Java Bean validation violation is for this particular execution metric.
+     * @return
+     */
+    public abstract String getPropertyPathToValidate();
+
+    /**
+     * Aggregates workflow executions into an aggregated metric and calculates the number of skipped executions.
      * @param executions
      * @return
      */
-    default List<E> getNonNullMetricsFromExecutions(List<T> executions) {
-        return executions.stream()
-                .map(this::getMetricFromExecution)
-                .filter(Objects::nonNull)
-                .toList();
+
+    public final Optional<M> getAggregatedMetricFromExecutions(List<T> executions) {
+        final List<T> executionsWithNonNullMetric = executions.stream().filter(execution -> getMetricFromExecution(execution) != null).toList();
+        final List<E> validExecutionMetrics = executionsWithNonNullMetric.stream().filter(this::isValid).map(this::getMetricFromExecution).toList();
+        if (!validExecutionMetrics.isEmpty()) {
+            Optional<M> calculatedMetric = calculateAggregatedMetricFromExecutionMetrics(validExecutionMetrics);
+            final int numberOfSkippedExecutions = executionsWithNonNullMetric.size() - validExecutionMetrics.size();
+            calculatedMetric.ifPresent(metric -> metric.setNumberOfSkippedExecutions(numberOfSkippedExecutions));
+            return calculatedMetric;
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Aggregates a list of aggregated metrics into one aggregated metric  nd calculates the number of skipped executions.
+     * @param aggregatedMetrics
+     * @return
+     */
+    public final Optional<M> getAggregatedMetricsFromAggregatedMetrics(List<M> aggregatedMetrics) {
+        if (!aggregatedMetrics.isEmpty()) {
+            Optional<M> calculatedMetric = calculateAggregatedMetricFromAggregatedMetrics(aggregatedMetrics);
+            // Sum number of skipped executions from the aggregated metrics
+            final int numberOfSkippedExecutions =  aggregatedMetrics.stream().map(Metric::getNumberOfSkippedExecutions).reduce(0, Integer::sum);
+            calculatedMetric.ifPresent(metric -> metric.setNumberOfSkippedExecutions(numberOfSkippedExecutions));
+            return calculatedMetric;
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Validate executions using Java Bean Validation
+     * @param execution
+     * @return
+     */
+    public boolean isValid(T execution) {
+        Set<ConstraintViolation<T>> violations = validator.validate(execution);
+        return violations.stream().noneMatch(violation -> violation.getPropertyPath().toString().startsWith(getPropertyPathToValidate()));
```

Comment:
Is the string-based comparison vulnerable to problems like `getPropertyPath().toString()` being `foobar` and `getPropertyPathToValidate()` being `foo`?  Violation has a `getNode` method containing the name, I think, might be safer use that.  In practice, probably doesn't matter tho.